code/ui/blocks/src/blocks/Subheading.tsx

```diff
@@ -8,7 +8,7 @@ export const Subheading: FC<PropsWithChildren<HeadingProps>> = ({ children, disa
   if (disableAnchor || typeof children !== 'string') {
     return <H3>{children}</H3>;
   }
-  const tagID = children.toLowerCase().replace(/[^a-z0-9]/gi, '-');
+  const tagID = children.toLowerCase().replace(/\s/gi, '-');
```

Comment:
We might want to utilize `toId` from `@storybook/csf` here instead to handle punctuation better.
Could you make the change in this PR, and see if that also adresses the original issue.

If it does not, could you extract this to a utility function and provides tests (including some with punctuation)